test/note_test.rb

```diff
@@ -2,19 +2,52 @@
 
 class TestNote < MiniTest::Test
   def setup
-    @frequency   = 220.to_f
-    @duration    = 2.0
-    @sample_rate = 22050
-    @note = MusicTheory::Note.new(frequency: @frequency, duration: @duration)
+    @test_samples = eval(File.read('test/fixtures/note_samples_array'))
+    @basic_note = MusicTheory::Note.new
+    @note_220 = MusicTheory::Note.new(frequency: 220)
+    @distorted_note = MusicTheory::Note.new(distort: true)
   end
 
-  def test_total_frames
-    frames = (@duration * @sample_rate).to_i
-    assert_equal frames, @note.total_frames
+  def test_samples_are_identical
+    assert sample_arrays_diff(@test_samples, @basic_note.samples) < (0.1 * (10 ** -10))
   end
 
-  def test_cycles_per_frame
-    cycles_per_frame = @frequency / @sample_rate
-    assert_equal cycles_per_frame, @note.cycles_per_frame
+  def test_number_of_samples_match
+    assert_equal @test_samples.length, @basic_note.samples.length
+    assert_equal @test_samples.length, @note_220.samples.length
+    assert_equal @test_samples.length, @distorted_note.samples.length
+  end
+
+  def test_number_of_frames_matches_number_of_samples
+    assert_equal @basic_note.total_frames, @basic_note.samples.length
+    assert_equal @note_220.total_frames, @note_220.samples.length
+    assert_equal @distorted_note.total_frames, @distorted_note.samples.length
+  end
+
+  def test_amplitude_range
+    assert each_array_item_less_than_1(@basic_note.samples)
+    assert each_array_item_less_than_1(@note_220.samples)
+    assert each_array_item_less_than_1(@distorted_note.samples)
+  end
+
+  private
+
+  def sample_arrays_diff(array_1, array_2)
+    diff = 0
+    zipped = array_1.zip(array_2)
+    zipped.each do |item|
+      diff += (item[0] - item[1]).abs
+    end
+    diff
+  end
+
+  def each_array_item_less_than_1(samples_array)
+    less_than_1 = true
```

Comment:
don't need this, if it makes it all the way through the loop you can just `return true` as the last line of the method